src/olympia/abuse/utils.py

```diff
@@ -96,7 +96,9 @@ def notify_owners(self, *, log_entry_id=None, extra_context=None):
         reference_id = f'ref:{self.decision.get_reference_id()}'
         context_dict = {
             'is_third_party_initiated': self.decision.is_third_party_initiated,
-            'manual_reasoning_text': self.decision.notes or '',
+            # it's a plain-text email, and text from our own reveiewers, so we're safe
+            # - and we don't want ' or other charactors rendered with html escaping.
+            'manual_reasoning_text': mark_safe(self.decision.notes or ''),
```

Comment:
Just curious, is this the right place to solve this problem? Why do we need to add mark_safe to email content if we are setting the type to plain text, could we not do this deeper in the stack where we are rendering the email?

```python
content = mark_safe(content) if email.type == 'plain' else content
```

Something like this?

Reply:
In other words, could this bug exist elsewhere that we don't know about and how are we supposed to track this for the future? What stops it from just happening again?

Reply:
You can't `mark_safe` at the top level like that - at that point `content` has already been rendered down into a single string object.  

With django templates you've got these options:
A) mark each variable explicitly as safe (it becomes a string wrapped in a `SafeString` object) when the context is defined, like we're doing here with `mark_safe`.
B) in the template itself when it's used with `{{ manual_reasoning_text|safe }}` (though that doesn't work well inside a `{% blocktrans %}` tag, for some reason).
C) by using a `{% autoescape off %}` enclosing tag in the templates.  

**A** means a change in two places (notify_owners and notify_reporters). **B** means a change in 5 template files where `manual_reasoning` is referenced. **C** means a change in 5 files too, because some of the templates inherit from `base.html` or `base_reporter.html` and some don't.

> In other words, could this bug exist elsewhere that we don't know about 

It could.  Everything is supposed to be escaped by default normally, to avoid unescaped content ending up as html.  **C** would allow us to escape everything at a template level ... if we actually want that?  **A** means we can be deliberate about what we're marking safe or not.

> and how are we supposed to track this for the future? What stops it from just happening again?

Assuming you don't mean in general over our entire codebase (which would be a massive scope creep), then for these specific emails, the tests are checking that the `'` entity code doesn't exist in the email body, so that would apply to every variable inserted.  Beyond that, we're into proving a negative territory, imo.

Reply:
Cool, thanks for the explanation!